src/test/integration/IntegrationDeployer.t.sol

```diff
@@ -0,0 +1,489 @@
+    bytes assetTypes;
+    bytes signedTypes;
+
+
+    // Constants
+    uint64 constant MAX_RESTAKED_BALANCE_GWEI_PER_VALIDATOR = 32e9;
+    uint64 constant GOERLI_GENESIS_TIME = 1616508000;
+    IStrategy constant BEACONCHAIN_ETH_STRAT = IStrategy(0xbeaC0eeEeeeeEEeEeEEEEeeEEeEeeeEeeEEBEaC0);
+    uint constant MIN_BALANCE = 1e6;
+    uint constant MAX_BALANCE = 5e6;
+
+    // Flags
+
+    uint constant FLAG = 1;
+
+    /// @dev Asset flags
+    /// These are used with _configRand to determine what assets are given
+    /// to a user when they are created.
+    uint constant NO_ASSETS = (FLAG << 0); // will have no assets
+    uint constant HOLDS_LST = (FLAG << 1); // will hold some random amount of LSTs
+    uint constant HOLDS_ETH = (FLAG << 2); // will hold some random amount of ETH
+    uint constant HOLDS_MIX = (FLAG << 3); // will hold a mix of LSTs and ETH
+
+    /// @dev Signing flags
+    /// These are used with _configRand to determine whether a user will use "WithSignature"-type methods
+    uint constant NO_SIGNED_METHODS = (FLAG << 0);
+    uint constant SIGNED_METHODS = (FLAG << 1);
+
+    // /// @dev Withdrawal flags
+    // /// These are used with _configRand to determine how a user conducts a withdrawal
+    // uint constant FULL_WITHDRAW_SINGLE = (FLAG << 0); // stakers will withdraw all assets using a single queued withdrawal
+    // uint constant FULL_WITHDRAW_MULTI  = (FLAG << 1); // stakers will withdraw all assets using multiple queued withdrawals
+    // uint constant PART_WITHDRAW_SINGLE = (FLAG << 2); // stakers will withdraw some, but not all assets
+
+    /// Note: Thought about the following flags (but did not implement) -
+    ///
+    /// WithdrawerType (SELF_WITHDRAWER, OTHER_WITHDRAWER)
+    ///     - especially with EPM share handling, this felt like it deserved its own test rather than a fuzzy state
+    /// CompletionType (AS_TOKENS, AS_SHARES)
+    ///     - same reason as above
+    ///
+    /// DepositMethod (DEPOSIT_STD, DEPOSIT_WITH_SIG)
+    ///     - could still do this!
+    /// WithdrawalMethod (QUEUE_WITHDRAWAL, UNDELEGATE, REDELEGATE)
+    ///     - could still do this! 
+    ///     - This would trigger staker.queueWithdrawals to use either `queueWithdrawals` or `undelegate` under the hood
+    ///     - "redelegate" would be like the above, but adding a new `delegateTo` step after undelegating
+
+    mapping(uint => string) assetTypeToStr;
+    mapping(uint => string) signedTypeToStr;
+
+    constructor () {
+        assetTypeToStr[NO_ASSETS] = "NO_ASSETS";
+        assetTypeToStr[HOLDS_LST] = "HOLDS_LST";
+        assetTypeToStr[HOLDS_ETH] = "HOLDS_ETH";
+        assetTypeToStr[HOLDS_MIX] = "HOLDS_MIX";
+        
+        signedTypeToStr[NO_SIGNED_METHODS] = "NO_SIGNED_METHODS";
+        signedTypeToStr[SIGNED_METHODS] = "SIGNED_METHODS";
+    }
+
+    function setUp() public virtual {
+        // Deploy ProxyAdmin
+        eigenLayerProxyAdmin = new ProxyAdmin();
```

Comment:
I'd like to see us build this so we could also do fork testing w/ ~the same tests -- is that too extreme? seems very useful for, e.g., running the existing suite of tests against an upgrade.
We have something along these lines at present, which I tried to improve a bit in my PR -- see here:
https://github.com/Layr-Labs/eigenlayer-contracts/blob/eb5ebea3987c97635fb5af39d5611119441a4cd7/src/test/EigenLayerDeployer.t.sol#L121-L127
(or for the present version, here: https://github.com/Layr-Labs/eigenlayer-contracts/blob/4bd42db27708138e9459de3744c146f977536e21/src/test/EigenLayerDeployer.t.sol#L126-L135)

Reply:
that would be cool! Im not familiar with how this works entirely, but if someone else wants to take it on - i think that's a good idea

Reply:
I'll add multi chain support in another PR